implementations/rust/ockam/ockam_api/src/authenticator/common.rs

```diff
@@ -0,0 +1,91 @@
+use crate::authenticator::direct::{OCKAM_ROLE_ATTRIBUTE_ENROLLER_VALUE, OCKAM_ROLE_ATTRIBUTE_KEY};
+use crate::authenticator::{AuthorityMember, AuthorityMembersRepository};
+use minicbor::Decoder;
+use ockam::identity::Identifier;
+use ockam_core::api::{RequestHeader, Response};
+use ockam_core::{Result, Routed};
+use ockam_node::Context;
+use std::collections::BTreeMap;
+use std::sync::Arc;
+
+/// Return a response on the return route stating that a secure channel is needed to access
+/// the service
+pub async fn secure_channel_required(c: &mut Context, m: Routed<Vec<u8>>) -> Result<()> {
+    // This was, actually, already checked by the access control. So if we reach this point
+    // it means there is a bug.  Also, if it' already checked, we should receive the Peer'
+    // identity, not an Option to the peer' identity.
+    let mut dec = Decoder::new(m.as_body());
+    let req: RequestHeader = dec.decode()?;
+    let res = Response::forbidden(&req, "secure channel required").to_vec()?;
+    c.send(m.return_route(), res).await
+}
+
+pub(crate) struct EnrollerCheckResult {
+    pub(crate) is_member: bool,
+    pub(crate) is_enroller: bool,
+    pub(crate) is_pre_trusted: bool,
+}
+
+pub(crate) struct EnrollerAccessControlChecks;
+
+impl EnrollerAccessControlChecks {
+    pub(crate) fn check_str_attributes_is_enroller(attributes: &BTreeMap<String, String>) -> bool {
+        attributes.get(OCKAM_ROLE_ATTRIBUTE_KEY)
+            == Some(&OCKAM_ROLE_ATTRIBUTE_ENROLLER_VALUE.to_string())
+    }
+
+    pub(crate) fn check_bin_attributes_is_enroller(
+        attributes: &BTreeMap<Vec<u8>, Vec<u8>>,
+    ) -> bool {
+        if let Some(val) = attributes.get(OCKAM_ROLE_ATTRIBUTE_KEY.as_bytes()) {
```

Comment:
Arguably a one-liner? 
```rust
attributes
    .get(OCKAM_ROLE_ATTRIBUTE_KEY.as_bytes())
    .map_or(false, |v| {
        v == OCKAM_ROLE_ATTRIBUTE_ENROLLER_VALUE.as_bytes()
    })
```

Reply:
Actually, I had a one-liner for `check_str_attributes_is_enroller`. But now I think that the long version is the only way to both avoid additional allocations during type conversion and make it simple